src/iceberg_functions/iceberg_multi_file_reader.cpp

```diff
@@ -195,6 +195,12 @@ void IcebergMultiFileList::InitializeFiles() {
 	auto manifest_list_full_path = options.allow_moved_paths
 	                                   ? IcebergUtils::GetFullPath(iceberg_path, snapshot.manifest_list, fs)
 	                                   : snapshot.manifest_list;
+
+	if (manifest_list_full_path.empty()) {
```

Comment:
I think we can check `snapshot.manifest_list.empty()` earlier (line 149), we don't need a manifest reader, manifest producer, entry producer etc.. when we're reading an empty table

Reply:
I also think this is leaving `current_data_manifest` and `current_delete_manifest` uninitialized, but they are being compared to `data_manifests.end()` and `delete_manifests.end()` respectively, which is undefined behavior (works on mac, won't work on linux)

Reply:
I think we want to set `snapshot_id` to `DConstants::INVALID_INDEX` and check for that to see if no snapshot could be found, before we do comparisons against uninitialized values (or make sure that these initialization steps do still run, but that's less future-proof I feel)